BinaryEncoding.md

```diff
@@ -201,9 +199,13 @@ invalidate a module.
 | name | `bytes` ? | section name string, present if `id == 0` |
 | payload_data  | `bytes` | content of this section, of length `payload_len - sizeof(name) - sizeof(name_len)` |
 
-Each known section is optional and may appear at most once.
-Custom sections all have the same `id`, and can be named non-uniquely (all bytes composing their names can be identical).
-Known sections from this list may not appear out of order.
+Each known section is optional and may appear at most once. Custom sections all have the same `id` (0), and can be named non-uniquely (all bytes composing their names may be identical).
+
+Custom sections are intended to be used for debugging information, future evolution, or third party extensions. For MVP, we use a specific custom section (the [Name Section](#name-section)) for debugging information. 
+
+If a wasm implementation reads and validates any custom section during module validation time, such validation errors do not invalidate the module.
+
+Known sections from the list below may not appear out of order, while custom sections may be interspersed before, between, as well as after any of the elements of the list, in any order. Certain custom sections may have their own ordering and cardinality requiremens. For example, the [Name section](#name-section) is expected to appear at most once, immediately after the Data section. 
```

Comment:
This may be confusing in combination with the preceding paragraph. Perhaps add a sentence to clarify that violation of such requirements may (only) cause an implementation to ignore the section.